### DISCUSSION TITLE
Suggest adding a bat file for convenience / Предлагаю добавить bat файл для удобства

### DISCUSSION THREAD

**Discussion by CinemaGame** [OnlyFart/Elib2Ebook#44] [Ideas]
I created it .bat file for convenience, so that you can download books entirely, just choosing the format and inserting a link to the desired resource, because every time I write that I call the Elib2Ebook application.exe is very annoying. Naturally, it is impossible to add additional parameters for the type of downloading of certain chapters, there is rather a question of convenience for downloading the whole book, but if necessary, I can make a more complex bat file that will take this into account, among other things

Создал я .bat файл для удобства, чтобы можно было целиком скачивать книги, просто выбирая формат и вставляя линк на нужный сайт, ибо каждый раз прописывать, что я вызываю(скорее уж призываю :D) приложение Elib2Ebook.exe очень раздражает. Естественно там нельзя добавить дополнительные параметры по типу скачивания определенных глав, тут скорее вопрос удобства для скачивания книги целиком, но если нужно, то могу сделать и более сложный бат файл, который будет учитывать это в том числе, но мне лень, да. Но как известно, лень - двигатель прогресса.

```
echo off
setlocal
cls

:perm_ask
echo Make a permanent format? 
CHOICE /C YN /N /T 15 /D N /M "Press Y for Yes and N for No: "
IF "%ERRORLEVEL%" == "1" set perm=TRUE
IF "%ERRORLEVEL%" == "2" set perm=FALSE

goto select_format

:select_format
cls
if "%perm%"=="TRUE" (echo You have selected the permanent save mode. Be CAREFUL! & echo.)

echo Select the file format:
echo =============
echo 1) .fb2 format
echo 2) .epub format
echo 3) .fb2 + .epub formats

CHOICE /C 123 /N /M "Format: "

IF "%ERRORLEVEL%" == "1" set form=fb2
IF "%ERRORLEVEL%" == "2" set form=epub
IF "%ERRORLEVEL%" == "3" set form=fb2,epub

goto url_insert

:url_insert
cls
if "%form%"=="fb2,epub" (echo You have chosen the .epub and .fb2 format
	) else (echo You have chosen the .%form% format)
echo. 
set /p url=Insert the URL: 

Elib2Ebook.exe -u %url% -f %form%

echo. 

CHOICE /C YN /T 10 /D Y /M "Continue? "

IF "%ERRORLEVEL%" == "1" goto select_format
IF "%ERRORLEVEL%" == "2" goto exit

:exit
@exit
```

**Comment by yarrg** [comment]
Спасибо, действительно удобно.

**Comment by WebEzhik** [comment]
Батник это хорошо, но предложенный слишком большой. Я вообще всегда качаю только полные книги в одном формате. А если вдруг будет нужно что то другое открою и допишу bat. Хотите быть ленивыми, будьте ими до конца. Вот что мне удобно:

```
echo off
setlocal
cls

set /p url=Insert the URL: 

Elib2Ebook.exe -u %url% -f epub

:exit
@exit
```
